lib/src/utils/schema/reducerActions.ts

```diff
@@ -0,0 +1,22 @@
+import { VernaSchemaType } from ':/types/rjsf';
+import { ReducerAction } from ':/types/VernaProvider';
+import { addSection, addWidget } from ':/utils/schema/adders';
+import { removeSection, removeWidget } from ':/utils/schema/removers';
+
+export function applyReducerAction(schema: VernaSchemaType, action: ReducerAction) {
+  switch (action.type) {
+    case 'addSection':
+      addSection(schema, action.payload?.id, action.payload.SectionTemplate);
+      break;
+    case 'addWidget':
+      addWidget(schema, action.payload?.id, action.payload?.widgetInfos);
+      break;
+    case 'removeSection':
+      removeSection(schema, action.payload?.id);
+      break;
+    case 'removeWidget':
+      removeWidget(schema, action.payload?.id);
+      break;
+  }
+  return schema;
+}
```

Comment:
I know this is something you are working.
Later all methods must be pures so we will have something like that :

```suggestion  let nextSchema = schema;
  switch (action.type) {
    case 'addSection':
      return addSection(schema, action.payload?.id, action.payload.SectionTemplate);
    case 'addWidget':
      return addWidget(schema, action.payload?.id, action.payload?.widgetInfos);
    case 'removeSection':
      return removeSection(schema, action.payload?.id);
    case 'removeWidget':
      return removeWidget(schema, action.payload?.id);
  }
}
```